crates/astria-sequencer/src/state_ext.rs

```diff
@@ -243,3 +243,298 @@ pub(crate) trait StateWriteExt: StateWrite {
+            .await
+            .unwrap();
+
+        // holds expected
+        let fee_balances_updated = state.get_block_fees().await.unwrap();
+        assert!(fee_balances_updated[0] == (asset, amount));
+    }
+
+    #[tokio::test]
+    async fn block_fee_read_and_increase_can_delete() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // can write
+        let asset_first = astria_core::sequencer::v1::asset::Id::from_denom("asset_0");
+        let asset_second = astria_core::sequencer::v1::asset::Id::from_denom("asset_1");
+        let amount_first = 100u128;
+        let amount_second = 200u128;
+
+        state
+            .get_and_increase_block_fees(asset_first, amount_first)
+            .await
+            .unwrap();
+        state
+            .get_and_increase_block_fees(asset_second, amount_second)
+            .await
+            .unwrap();
+        // holds expected
+        let fee_balances = state.get_block_fees().await.unwrap();
+        for val in fee_balances {
+            assert!(val == (asset_first, amount_first) || val == (asset_second, amount_second));
+        }
+
+        // can delete
+        state.clear_block_fees().await;
+
+        let fee_balances_updated = state.get_block_fees().await.unwrap();
+        assert!(fee_balances_updated.is_empty());
+    }
+
+    #[tokio::test]
+    async fn is_allowed_fee_asset() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // non-existent fees assets return false
+        let asset = astria_core::sequencer::v1::asset::Id::from_denom("asset_0");
+        assert!(!state.is_allowed_fee_asset(asset).await.unwrap());
+
+        // existent fee assets return true
+        state.put_allowed_fee_asset(asset);
+        assert!(state.is_allowed_fee_asset(asset).await.unwrap());
+    }
+
+    #[tokio::test]
+    async fn can_delete_allowed_fee_assets_simple() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // setup fee asset
+        let asset = astria_core::sequencer::v1::asset::Id::from_denom("asset_0");
+        state.put_allowed_fee_asset(asset);
+        assert!(state.is_allowed_fee_asset(asset).await.unwrap());
+
+        // see can get fee asset
+        let assets = state.get_allowed_fee_assets().await.unwrap();
+        assert!(assets[0] == asset);
+
+        // can delete
+        state.delete_allowed_fee_asset(asset);
```

Comment:
check what happens if you delete a fee asset that wasn't stored - should error

Reply:
Right now `delete_allowed_fee_asset()` doesn't ensure that the fee asset exists before deleting. Do you want me to change the behavior to err in this case? I can make a follow up PR for it

Reply:
you can leave as-is, thinking about it it doesn't really make a difference if this errors or is a no-op